Enhances remote provider connection flow

sergeibbb opened the pull request:
# Description

This is a follow-up on #4387

Originates from [this conversation](https://gitkraken-hq.slack.com/archives/C06QFFUSVUZ/p1750157934439699?thread_ts=1750096212.425849):

> And there is another thing that I think is worth fixing: handling of the "connect" click should be different. I'm attaching the example how it's handled in Remotes View and in Home. I think on Home it should work the same way as on Remotes View.

How it should work (without a web page, using existing connection):

https://github.com/user-attachments/assets/7fb47859-2025-4782-90cc-716606009293

How it works (it passes user through a web page) and what this pull request changes:

![image](https://github.com/user-attachments/assets/542b07c8-fb09-4328-8ea9-f99885ebf477)


# Verification steps

- [ ] basic change: open "remotes", disconnect, open "Home", connect: it should not go through the browser.
- [ ] right remote is selected
  - switch to a branch, that has an upstream from `origin`
    <img width="300" height="84" alt="image" src="https://github.com/user-attachments/assets/66816656-79a9-4e28-96a1-4e86f505c062" />

  - on remotes: disconnect, if there is a default remote then "Unset as default"
  - open "Home", connect.
  - [ ] on "remotes" check that the `origin` is selected as default
  - edit the branch, change its upstream to another remote
    <img width="300" height="86" alt="image" src="https://github.com/user-attachments/assets/37cddf3a-7712-4bd2-a169-e79c79d333bf" />
  - disconnect and "Unset the default" again
  - open "Home", connect
  - [ ] on "remotes" check that the `orther` is set as default
- [ ] prioritized remote is selected
  - switch to a branch without an upstream
    <img width="300" height="74" alt="image" src="https://github.com/user-attachments/assets/34e642b2-b974-4a5d-b72d-6dd5744850f3" />
  - Have a remote called as "upstream"
  - disconnect and "Unset the default"
  - open "Home", connect
  - [ ] on "remotes" check that no "upstream" is set as default
  - [ ] same steps but no "upstream" remote: "origin" is set as default


# Checklist

<!-- Please check off the following -->

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [ ] My changes include any required corresponding changes to the documentation (including CHANGELOG.md and README.md)
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

sergeibbb commented:
@axosoft-ramint What do we do with this? Reject, accept, transform to different requirements?

cc @eamodio

axosoft-ramint commented on src/git/models/repositoryShape.ts:
```diff
@@ -14,5 +14,7 @@ export interface RepositoryShape {
 		integration?: { id: SupportedCloudIntegrationIds; connected: boolean };
 		supportedFeatures: RemoteProviderSupportedFeatures;
 		url?: string;
+		remotePath?: string;
+		remoteName?: string;
```
I see `remotePath` being added here but I don't see where it is being used?

Maybe call it `bestRemoteName` since in the only two places we use this, it's the so-called "best remote" of the repo?

sergeibbb replied:
Fixed

axosoft-ramint commented on src/webviews/apps/shared/components/repo-button-group.ts:
```diff
@@ -198,9 +198,9 @@ export class GlRepoButtonGroup extends GlElement {
-										{ integrationIds: [provider.integration!.id], source: this.source },
+									href=${createCommandLink<ConnectRemoteProviderCommandArgs>(
+										'gitlens.connectRemoteProvider',
+										{ remote: provider.remoteName!, repoPath: repo.path },
```
Is this guaranteed to be the remote that the current branch is located on since this is where the link is located?

sergeibbb replied:
@axosoft-ramint 

This is a good question. For picking the right integration the remote is unnecessary, because the only thing that matters is the integration type.

The problem here is setting the default remote when the integration is connected.

The old flow, that goes through the web site (ConnectCloudIntegrationsCommand), does not set the default remote when the integration is connected.
In the new flow the default remote is set to the value of the parameter that might be wrong.

So, I suggest to make the `remote` field optional and if it's not set, then do not set the default remote on connecting an integration. By doing that we:
1. keep the old behavior (not setting the remote at all)
2. won't fail with setting a wrong remote as default.

What do you think?

sergeibbb replied:
@axosoft-ramint 

This is a good question. For picking the right integration the remote is unnecessary, because the only thing that matters is the integration type.

The problem here is setting the default remote when the integration is connected.

The old flow, that goes through the web site (ConnectCloudIntegrationsCommand), does not set the default remote when the integration is connected.
In the new flow the default remote is set to the value of the parameter that might be wrong.

So, I suggest to make the `remote` field optional and if it's not set, then do not set the default remote on connecting an integration. By doing that we:
1. keep the old behavior (not setting the remote at all)
2. won't fail with setting a wrong remote as default.

What do you think?

sergeibbb commented:
Hi @axosoft-ramint 

I've removed passing the remote. Now it's defined by some logic that already exists.
Also I've added "Verification steps" that describes how it selects the default origin and how it can be tested.
